app/src/processing/app/Base.java

```diff
@@ -1376,16 +1376,29 @@ public void onBoardOrPortChange() {
       }
     }
     newRecentBoardIds.add(0, currentBoard);
-    if (newRecentBoardIds.size() == 6) {
-      newRecentBoardIds.remove(5);
+
+    int numBoards = 0;
+
+    if (PreferencesData.has("recent.num_boards")) {
+      numBoards = PreferencesData.getInteger("recent.num_boards");
+    }
+
+    while (newRecentBoardIds.size() > numBoards) {
+      newRecentBoardIds.remove(newRecentBoardIds.size() - 1);
     }
     PreferencesData.setCollection("recent.boards", newRecentBoardIds);
-    try {
-      rebuildRecentBoardsList();
-    } catch (Exception e) {
-      //TODO show error
-      e.printStackTrace();
-    }    
+
+    // If recent.num_boards is 0, interpret this as the feature
+    // being turned off. There's no need to rebuild the menu
+    // because it will be hidden
+    if (numBoards > 0) {
```

Comment:
I believe this if could be broader: If `numBoards == 0`, then there's no point to even set up, add to and prune the list at all. It might still be needed to create an empty list to keep other code happy, but maybe not even.

Reply:
In my branch, I ended up just removing this `if` instead, since the code should produce the right result in that case anyway (it is slightly slower, but having this disabled is a corner case that should not really be optimized for).